### DISCUSSION TITLE
Experimental library site using OSD: Now in beta

### DISCUSSION THREAD

**Discussion by DevinSmithWork** [openseadragon/openseadragon#2551] [General]
Hello all!

I'm at the tail-end of building a large project for a small, experimental library here in San Francisco.

Among other idiosyncrasies, this library doesn't keep a formal catalog -- hence we decided to build a photo-based interface emphasizing the visual browsing experience rather than the search-box-first approach of most library websites. I'm finishing up the much-improved second version, which now runs proudly on OpenSeadragon: [Prelinger Library Stacks Explorer v2 (beta)](http://prelingerlibrary.org/stacks/v2)

If anyone has any feedback on this, I'd love to hear it.

In terms of completeness, I'd say the coding is 90% done; Most of the remaining work is data entry & cleanup (Only about 1/3 of the Internet Archive scanned material is linked so far). And of course fixing typos ("Braodcasting").

Lastly, I'd like to say a big thank you to @iangilman for your responsiveness on the Q&A forum here. You've definitely saved me several hours & headaches while building this project.

Cheers!
- Devin

**Comment by msalsbery** [comment]
Looks great!

**Comment by pearcetm** [comment]
Very cool! I see that certain info updates in the left panel when you've clicked on something (if the menu is expanded), and there's also a pulsing green circle at the location of the click. The info in the left panel doesn't seem related to the actual book that is clicked on (just the overall section?) - but maybe this is because of the full detailed info isn't finished?

Regardless, from a user experience perspective, I have a couple of thoughts for you (since you asked for feedback :) )

- It wasn't immediately intuitive that I'd need to click on "Menu" to expand this and show info of what I clicked on. Automatically opening this on a click action could be helpful (maybe with a checkbox to opt out of auto-open on future clicks).
- The pulsing green circle was confusing until I knew that there was related info shown in the left panel
- Clicking in OpenSeadragon by default triggers a zoom action, which complicates using the same interaction to select a region for extra info to be displayed. Is click-to-zoom needed? Probably in some circumstances, if there's no scroll wheel or pinch-to-zoom... but maybe click-to-zoom could be an "opt-in" feature, or autodetect if it is needed somehow? 
- Displaying the detailed info in-place (rather than in the left panel) as part of a right-click context menu could be more intuitive, though for devices without ready access to context menus you'd want a fallback still.

**Reply by DevinSmithWork** [reply]
Thank you so much for the feedback!

I'd been considering some UI changes along these lines, so your comment is good confirmation. This week I'm going to play around with (1) changing the click indicator circle to highlighting the whole shelf, and (2) adding a button beside the canvas click location that'll open a div with the "you clicked here" info from the menu.

Thanks again for taking a look!

**Comment by iangilman** [comment]
This is fantastic! I love the feeling of actually browsing through real bookshelves. 

I really want to be able to click on a book and have it take me right into reading it. Looks like that is possible with some books, but not others? I assume some of this is just because it's not finished.

Anyway, I love it! What a delightful use for the technology ❤️

**Reply by DevinSmithWork** [reply]
Thank you! And thanks again for all your helpful answers with my technical Qs!

Unfortunately, we don't have every book in the library scanned (the scanned collection is only ~5k items). The strategy we settled on was "shelving" the scanned items into particular zones+subzones; Then when a user clicks on a shelf, I pull up a list of all the scanned items "shelved" on that shelf's same zone+subzone.

There's some pros-and-cons to this strategy: The main advantage is that we can shelve a digital book in more than one zone+subzone. The best example of this is education books for kids. Typically these would be physically shelved together, but now we can shelve things like "Don't play in the street!" both in the kids' section and in the (adult) automobility section.

(A lot of these unusual decisions come from the library's lack of a catalog; in a more formal library, we could (e.g.) encode a range of call numbers for a particular shelf, and use that metadata to more precisely surface individual scanned titles.)

Adding some kind of on-canvas window w/ the scanned item list next to the click location seems to be what most people would like, so I'll be implementing that soon.

Cheers!

**Reply by iangilman** [reply]
Cool. You've got to work with what you've got!

But yeah, since it's not what the user would expect, it's important to educate them about how your system works. Part of that is presenting information more directly, rather than having it hidden under the "menu" button.

I look forward to seeing how it evolves!

**Comment by DevinSmithWork** [comment]
Hi all, I've been steadily making revisions & improvements, and am now in the home stretch: [https://prelingerlibrary.org/stacks/v2/](https://prelingerlibrary.org/stacks/v2/)

- Data entry completed for scanned items
- Rewrote the local data loader for stability
- Menu is now default-open, and divided into "click view" and "lists view." This should ideally alleviate some of the confusion around where the info about the clicked-on stacks can be found (I heard this same feedback not just here but with all my other user groups).
- The list view's "zones/subzones" now aggregates the scanned items at the subzone rather than zone level.
- Scanned items now have illustration-level indicators (1 to 3 green diamonds)
- Clicking on a shelf no longer auto zooms; added double-click to zoom in.
- Short UI animations added to draw the eye toward elements being changed or highlighted.
- A bunch of under-the-hood cleanup and general bolt-tightening.

The only major work left is adding CSS media queries for mobile, and writing the help/documentation pages.

I'd love to hear if y'all have any more feedback! @iangilman @pearcetm @msalsbery 

Cheers!
Devin

**Comment by iangilman** [comment]
Gorgeous! Having the menu open makes a big difference. This is great fun to explore :)

This would be a great project to add to https://openseadragon.github.io/examples/in-the-wild/. If you'd like to do the honors, you can make a PR against this file: https://github.com/openseadragon/site-build/blob/master/www/in-the-wild.html

Congrats on a lovely experience!

**Reply by DevinSmithWork** [reply]
Thank you! I would be honored to add this to the examples page; I have a small handful of cleanups to execute over the weekend, and will put in the PR after that.

Cheers!

**Reply by iangilman** [reply]
Awesome 😊

**Comment by pearcetm** [comment]
It's looking great!

As I was poking around, there were a couple of things I noticed about the interactions with the image and the list that you could consider:

- Clicking on some items in the list add highlights to the image, while others open a new tab. Distinguishing these two actions visually some how (cursor style? link color?) would help the behavior be more predicable as users are browsing around
- "Clear Highlights" also collapses the list, instead of just deselecting the current selections, which felt strange to me.
- I felt like it would be nice to clear the highlights of sub-items when collapsing a subzone.
- When clicking on a shelf, it wasn't intuitive to me why items in the entire subzone are listed on the left. Perhaps adding the subzone graphical indicator on the image would help, if it isn't feasible to only list those that are on the selected shelf.
- In the "list" view (as opposed to "click" view) it would be nice if clicking a shelf added highlights to the list items that correspond to the subzone and shelf.

These are just suggestions, of course - it is already really cool! Congratulations!
